### DISCUSSION TITLE
Paginator

### DISCUSSION THREAD

**Discussion by weizhongzou** [C-Accel-CRIPT/Python-SDK#181] [Q&A]
What’s the role of “paginator”?

**Comment by nh916** [comment]
Paginator is essentially a way for us to flip through the different results from the API. If we ask for materials from the API and it has 5 million materials, it doesn’t return all 5 million at once and instead just returns the first 10, if we want more, we can flip to the second page for the next 10 results and so on. Paginator is an easy way for the user to do that without having to know what URL parameters and different things that are needed to flip through the results. They can just do paginator.next() to go to the next page.

**Reply by InnocentBug** [reply]
we have the plan to make this transparent for the user at some point using the paginator like a python generator.
